diff --git a/include/compiz-core.h b/include/compiz-core.h
index c4ed676..652a319 100644
--- a/include/compiz-core.h
+++ b/include/compiz-core.h
@@ -979,6 +979,7 @@ struct _CompDisplay {
     Atom clientListStackingAtom;

     Atom frameExtentsAtom;
+    Atom frameGtkExtentsAtom;
     Atom frameWindowAtom;

     Atom wmStateAtom;
diff --git a/src/display.c b/src/display.c
index 8e16e5e..7873b16 100644
--- a/src/display.c
+++ b/src/display.c
@@ -2115,6 +2115,7 @@ addDisplay (const char *name)
 	XInternAtom (dpy, "_NET_CLIENT_LIST_STACKING", 0);

     d->frameExtentsAtom = XInternAtom (dpy, "_NET_FRAME_EXTENTS", 0);
+    d->frameGtkExtentsAtom = XInternAtom (dpy, "_GTK_FRAME_EXTENTS", 0);
     d->frameWindowAtom  = XInternAtom (dpy, "_NET_FRAME_WINDOW", 0);

     d->wmStateAtom	  = XInternAtom (dpy, "WM_STATE", 0);
diff --git a/src/screen.c b/src/screen.c
index 443f4f0..d68d544 100644
--- a/src/screen.c
+++ b/src/screen.c
@@ -1234,6 +1234,7 @@ addSupportedAtoms (CompScreen   *s,

     atoms[count++] = d->wmUserTimeAtom;
     atoms[count++] = d->frameExtentsAtom;
+    atoms[count++] = d->frameGtkExtentsAtom;
     atoms[count++] = d->frameWindowAtom;

     atoms[count++] = d->winStateAtom;
